crates/agglayer-interop-types/src/bincode.rs

```diff
@@ -0,0 +1,93 @@
+pub use bincode::{Error, Options, Result};
+
+/// Bincode configuration. Deliberately inaccessible from the outside.
+mod options {
+    use bincode::config::{
+        AllowTrailing, BigEndian, Bounded, DefaultOptions as BincodeDefaultOptions, FixintEncoding,
+        Options as _, WithOtherEndian, WithOtherIntEncoding, WithOtherLimit, WithOtherTrailing,
+    };
+
+    pub type Default =
+        WithOtherIntEncoding<WithOtherEndian<BincodeDefaultOptions, BigEndian>, FixintEncoding>;
+
+    #[inline]
+    pub fn default() -> Default {
+        bincode::options().with_big_endian().with_fixint_encoding()
+    }
+
+    pub type Limited<T> = WithOtherLimit<T, Bounded>;
+
+    pub type SP1v4 = WithOtherTrailing<
+        WithOtherIntEncoding<BincodeDefaultOptions, FixintEncoding>,
+        AllowTrailing,
+    >;
+
+    #[inline]
+    pub fn sp1v4() -> SP1v4 {
+        bincode::options()
+            .with_fixint_encoding()
+            .allow_trailing_bytes()
+    }
```

Comment:
just for my understanding, where this `sp1v4` option is meant to be used?

The only bincode options that I have used so far, like for the public inputs of the PP program, I used `with_fixint_encoding` and `with_big_endian` [here](https://github.com/agglayer/agglayer/blob/9e38232a55ca47356910b5e3b7ae46c6329020bc/crates/agglayer-storage/src/columns/mod.rs#L19-L23)

(In `provers` also that's `with_fixint_encoding` + `with_big_endian` in some places)

Reply:
> just for my understanding, where this `sp1v4` option is meant to be used?

It's what sp1 uses to encode/decode the data passed to `SP1Stdin`. Also if you use plain `bincode::{encode, decode}`, it uses these settings for some reason.

Reply:
> The only bincode options that I have used so far, like for the public inputs of the PP program, I used `with_fixint_encoding` and `with_big_endian` [here](https://github.com/agglayer/agglayer/blob/9e38232a55ca47356910b5e3b7ae46c6329020bc/crates/agglayer-storage/src/columns/mod.rs#L19-L23)

Introduced the `bincode::contracts` option set to cover this use case.